library-best-practices/glossary.md

```diff
@@ -3,29 +3,38 @@ Java Dependency Glossary
 
 ### Types of conflicts and compatibility
 
-- **Linkage error**: an error when a Java class in a classpath has reference to
-  another class file for a class, field or method, and the reference can not be
-  satisfied with the available classes in the classpath.
+- **Linkage error**: an error when a Java class in a classpath references
+  another class (for the class itself, its field, or its method), and the reference
+  cannot be satisfied with the available classes in the classpath.
   A linkage error is caused by a reference to missing classes or a linkage conflict.
   Linkage errors detected at runtime manifest as `ReflectiveOperationException`,
   `NoClassDefFoundError`, `NoSuchFieldException`, `MethodNotFoundException`,
   `LinkageError`, or other related exceptions.
   - Sub-type: **Linkage conflict**
-
-- **Linkage conflict**: The signature, return type, modifiers, or throws
-  declaration of a non-private method or class in a dependency has changed in an
-  incompatible way between the version supplied at compile time and the version
-  invoked at runtime. For example, a public method may be removed from a class
-  or a class may be made final.
-  - Or, another perspective: In cases where binary compatibility and source
-    compatibility are the same, a linkage conflict is when compilation would
-    fail if the libraries in the classpath were all built together from their
-    originating source code, or when reflection would fail.
-  - Opposite: **Linkage-compatible**.
-  - Sub-type: **Static linkage conflict**: A linkage conflict caused by a direct
-    code reference (non-reflective).
-  - Sub-type: **Reflective linkage conflict**: A linkage conflict caused by a
-    reflective reference.
+  - Sub-type: **Missing class error**
+
+  - **Linkage conflict**: The signature, return type, modifiers, or throws
+    declaration of a non-private method or class in a dependency has changed in an
+    incompatible way between the version supplied at compile time and the version
+    invoked at runtime. For example, a public method may be removed from a class
+    or a class may be made final.
+    - Or, another perspective: In cases where binary compatibility and source
+      compatibility are the same, a linkage conflict is when compilation would
+      fail if the libraries in the classpath were all built together from their
+      originating source code, or when reflection would fail.
+    - Opposite: **Linkage-compatible**.
+    - Sub-type: **Static linkage conflict**: A linkage conflict caused by a direct
+      code reference (non-reflective).
+    - Sub-type: **Reflective linkage conflict**: A linkage conflict caused by a
```

Comment:
Generally "dynamic" is the opposite of "static".

Reply:
Updated